test/controllers/APIControllerSpec.scala

```diff
@@ -0,0 +1,138 @@
+package controllers
+
+import DAO.BakeryDB
+import org.scalatestplus.play._
+import org.scalatestplus.play.guice.GuiceOneAppPerTest
+import play.api.test.Helpers._
+import play.api.test.{FakeRequest, Injecting}
+import play.api.mvc._
+import java.time.LocalDate
+import java.util.UUID
+import scala.concurrent.Future
+import scala.language.postfixOps
+
+class APIControllerSpec(bakeryDB: BakeryDB)
+    extends PlaySpec
+    with Results
+    with GuiceOneAppPerTest
+    with Injecting {
+
+  "Create product" should {
+    "Return successful" in {
+      val controller = inject[APIController]
+      val result: Future[Result] = controller
+        .postProduct()
+        .apply(
+          FakeRequest(
+            POST,
+            "/pass-bakery/product"
+          ).withTextBody(
+            text =
+              "{\"name\": \"crepe\", \"quantity\": \"5\", \"price\": \"4.99\"}"
+          )
+        )
+
+      val bodyText: String = contentAsString(result)
+      bodyText mustBe ("Post successful, 1 rows updated")
+    }
+    "Return unsuccessful" in {
+      intercept[Exception] {
+        val controller = inject[APIController]
+        val result: Future[Result] = controller
+          .postProduct()
+          .apply(
+            FakeRequest(
+              POST,
+              "/pass-bakery/product"
+            ).withTextBody(
+              text = "{}"
+            )
+          )
+      }
+    }
+  }
+  "Edit product" should {
+    "Return successful" in {
+      val controller = inject[APIController]
+      val uuid: UUID = UUID.fromString("a62bf2f7-732d-47a0-b791-3140784784b0")
+      val result: Future[Result] = controller
+        .editByID(uuid)
+        .apply(
+          FakeRequest(
+            PUT,
+            " http://localhost:9000/rest/bakery/product/a62bf2f7-732d-47a0-b791-3140784784b0"
+          ).withTextBody(
+            text = "{\"name\": \"crepe\", \"price\": \"9.99\"}"
+          )
+        )
+
+      val bodyText: String = contentAsString(result)
+      bodyText mustBe ("1")
+    }
+    "Return unsuccessful" in {
+
+      val controller = inject[APIController]
+      val uuid: UUID = UUID.fromString("a62bf2f7-732d-47a0-b791-3140784704b0")
+      val result: Future[Result] = controller
+        .editByID(uuid)
+        .apply(
+          FakeRequest(
+            PUT,
+            "/pass-bakery/product"
+          ).withTextBody(
+            text = "{\"name\": \"crepe\", \"price\": \"9.99\"}"
+          )
+        )
+      val bodyText: String = contentAsString(result)
+      bodyText mustBe ("Product not found")
+    }
+  }
+  "findAllProducts" should {
+    "Should return success" in {
+      val controller = inject[APIController]
+      val result: Future[Result] = controller
+        .findAllProducts()
+        .apply(FakeRequest(GET, "/rest/bakery"))
+      val bodyText: String = contentAsString(result)
+      status(result) mustBe 200
+    }
+  }
+  "Status endpoint" should {
+    "Should return status" in {
```

Comment:
not a useful description. people who call the endpoint know what the endpoint will return. 

status endpoint should return the application status on GET call